modules/ossm-migrating-gateways-canary.adoc

```diff
@@ -0,0 +1,87 @@
+//
+// * service-mesh-docs-main/migrating/migrating-gateways/ossm-migrating-gateways-assembly.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-migrating-gateways-canary_{context}"]
+= Gateway canary migration
+
+Use the gateway canary migration method when you want a gradual rollout. It uses multiple gateway versions with maximum control over your gateway rollout.
+
+[NOTE]
+====
+The label for the gateway namespace differs between mulitenant and cluster-wide meshes. To understand which labels to use in your specific migration case and the `maistra.io/ignore-namespace: "true"` label requirement, see the "Multitenant migration guide" or the "Cluster-wide migration guide" section.
+====
+
+.Procedure
+
+. Run the following command to label the gateway namespace to ensure that gateway injection is enabled from the new mesh and add the `maistra.io/ignore-namespace: "true"` label:
++
+[source,terminal]
+----
+$ oc label namespace <gateway_namespace> istio.io/rev=<istio_revision_name> maistra.io/ignore-namespace="true"
+----
++
+.. Remove the `istio-injection=enabled` label, if needed.
+
+. Deploy a canary gateway by using the following example:
++
+.Example YAML for canary gateway
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway-canary
+  namespace: istio-ingress <1>
+  spec:
+     selector:
+       matchLabels:
+         istio: ingressgateway <2>
+     template:
+       metadata:
+         annotations:
+           inject.istio.io/templates: gateway
+         labels:
+           istio: ingressgateway
+           istio.io/rev: canary <3>
+       spec:
+         containers:
+         - name: istio-proxy
+           image: auto
+----
+<1> Your `Deployment` resource must be in in the same namespace as your existing gateway.
+<2> Your `spec.selector.matchlabels.istio` must match your existing gateway service selector.
+<3> Set your {SMProduct} 3.0 control plane revision as the value of the `istio.io/rev` label.
+
+. Ensure that the new gateway deployment is running with the new revision and is handling requests:
++
+.. Check that pods are running and in the `Ready` status.
+
+.. Check that the gateway is running the new revision by running the following command:
++
+[source,terminal]
+----
+$ oc istioctl ps -n istio-ingress
+----
+
+.. Test a sample route through the gateway.
+
+. Gradually shift traffic between deployments:
++
+.. Increase replicas for new gateway by running the following command:
++
+[source,terminal]
+----
+$ oc scale -n istio-ingress deployment/<new_gateway_deployment> --replicas <new_number_of_replicas>
+----
+
+.. Decrease replicas for old gateway by running the following command:
+[source,terminal]
++
+----
+$ oc scale -n istio-ingress deployment/<old_gateway_deployment> --replicas <new_number_of_replicas>
+----
+
+. Repeat step 4, incrementally adjusting replica counts until the new gateway handles all traffic to the gateway Service.
```

Comment:
While I couldn't find a style guide rule on this, I think it's generally best to avoid referencing specific steps, especially by number. If you had to keep this the same way, you could rephrase this as "Repeat the previous step", but that this encounters the same issues as referencing a number IMO (accuracy breaking if steps are added in this procedure).

Since this is referencing the previous steps, I think this could be simply be rewritten to be a part of step 4:
```suggestion
.. Repeat this process, incrementally adjusting replica counts until the new gateway handles all traffic to the gateway Service.
```